### PR TITLE
Fix JavaScript static type errors

### LINKED ISSUE #5878: JavaScript static type errors
### Description

Follow-up to: Slack convo linked below
Fix our current 18 JS static type errors detected running our TypeScript compiler

### Screenshots

screenshot_2023-03-01_at_10 57 25-2

### To Reproduce

```sh
yarn tsc # locally
tsc # globally, nb: if installed, likely a diff ver so results may vary
```

### Expected behavior

No JS type errors

### **Technical Details**

Currently we rely on IDE's (e.g. VSCode) to flag type errors during development and bump into these in runtime

JavaScript and JSX files cannot parse types (yet, see 2 refs below)

Next: Address remaining TypeScript errors

### **References**

- https://consensys.slack.com/archives/G1L7H42BT/p1677193361291439
- https://consensys.slack.com/archives/C02U025CVU4/p1677256019459879

+ https://devblogs.microsoft.com/typescript/a-proposal-for-type-syntax-in-javascript
+ https://github.com/tc39/proposal-type-annotations

- https://github.com/MetaMask/metamask-mobile/issues/4605

-------------------------------------------------------------

**Severity**

- Is this tech debt? ✅

---

cc @Cal-L @gantunesr

### PR DESCRIPTION
**Pull request by leotm**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

Fix: https://github.com/MetaMask/metamask-mobile/issues/5878

**Checklist**

* [x] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

cc @Cal-L @gantunesr

### CODE REVIEW

**Comment by leotm** [comment]
didn't initially see the 2 failing tests, taking a look

**Comment by gantunesr** [comment]
Hey @leotm, looks like this PR has a lot of linting errors

**Comment by leotm** [comment]
> Hey @leotm, looks like this PR has a lot of linting errors

yep spotted taking a look ^ our husky pre-commit hook running lint-staged is only running `eslint` not `yarn lint`, i'll PR too after

**Comment by leotm** [comment]
as i'm renaming (prefixing, we could also suffix) our ~50ish destructed core>engine controller variables from context

Screenshot 2023-03-03 at 18 21 49

due to duplicate names from upper scope (imports) - not necessarily a bad thing

Screenshot 2023-03-03 at 18 22 57

are these renames something we prefer? as we continue refactoring from JS to TS and restore our type checking (TSC) while fixing type errors...

or happy to disable `@typescript-eslint/no-shadow` (along with `no-shadow` currently disabled)

some food for thought for Monday i'll bring up then ^

**Comment by leotm** [comment]
after some thought it's not worth reducing strictness and disabling `@typescript-eslint/no-shadow` for our entire codebase

so only disabled it for `Engine.ts` for now until decided whether we want to go ahead with aforementioned var renames which will take considerable time

File: app/core/Engine.ts [hunk 1]
```diff
@@ -776,9 +773,9 @@ class Engine {
 
     // Recreate imported accounts
     if (importedAccounts) {
-      for (let i = 0; i < importedAccounts.length; i++) {
+      for (const importedAccount of importedAccounts) {
         await KeyringController.importAccountWithStrategy('privateKey', [
-          importedAccounts[i],
+          importedAccount,
```
Comment (leotm) [inline app/core/Engine.ts:778]: this appear correct from my understanding working w/o current type safety

unless should be e.g.

```
for (const id of importedAccounts) {
        await KeyringController.importAccountWithStrategy('privateKey', [
          importedAccounts[id],
```

**Comment by leotm** [comment]
thanks @sethkfman for re-actioning CI ^ i'll fix the non-components tests then re-ready up

**Comment by leotm** [comment]
fixed the cause of the non-component tests in https://github.com/MetaMask/metamask-mobile/pull/5879/commits/a4951879281ab1d690d76ee2248ea396042a6cb6

**Comment by leotm** [comment]
after fixing the tests, i'm still getting this for 1 test, even on `main` after fresh `git clean -fdx` and `yarn setup` 🤔
any ideas?

Screenshot 2023-03-06 at 17 45 39

**Comment by leotm** [comment]
Harry's now actioned my GitHub stuff 🎉 unsure @sethkfman why still requiring you to action some CI 🤔

Screenshot 2023-03-06 at 17 48 01

maybe i need to close this and re-create the PR, can do this when i'm back

but still wondering about the 1 failing auth test getting locally on `main` https://github.com/MetaMask/metamask-mobile/pull/5879#issuecomment-1456632738

**Comment by leotm** [comment]
> Add more commits by pushing to the [fix/javascript-static-type-errors](https://github.com/leotm/metamask-mobile/tree/fix/javascript-static-type-errors) branch on [leotm/metamask-mobile](https://github.com/leotm/metamask-mobile).

Closing in favour of new PR, old fork no longer exists, pushing/publishing no longer working and:

`gh pr checkout 5879`

> From https://github.com/MetaMask/metamask-mobile
 ! [rejected]            refs/pull/5879/head -> fix/javascript-static-type-errors  (non-fast-forward)
failed to run git: exit status 1

**Pull request closed without merging**